src/appengine/handlers/jobs.py

```diff
@@ -44,6 +53,30 @@ def get_queues():
   return queues
 
 
+def get_results(this):
+  """Get results for the jobs page."""
+  params = dict(this.request.iterparams())
+  # Return jobs sorted alphabetically by name
+  query = datastore_query.Query(data_types.Job)
+  query.order('name', is_desc=False)
+  filters.add(query, params, FILTERS)
+
+  page = helpers.cast(
```

Comment:
probably makes sense to check that the value is not negative

Reply:
Yes, actually I have opened a issue(#1883) for the same. This issue exists everywhere in testcase-list, crash and here as well. I was actually planning to add a fix for this datastore_query.Query class' fetch_page method, as this would ensure that it will be fixed everywhere and we wouldn't have to add explicit validations in all pages.

Reply:
sg!